src/datetime.rs

```diff
@@ -16,6 +16,30 @@ use Date;
 use format::{Item, Numeric, Pad, Fixed};
 use format::{parse, Parsed, ParseError, ParseResult, DelayedFormat, StrftimeItems};
 
+/// Specific formatting options for seconds
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum SecondsFormat {
```

Comment:
Actually: we should probably name this `FractionFormat` and also add a [`#[doc(hidden)] __Nonexhaustive` variant](https://github.com/diesel-rs/diesel/blob/5435e30f1dc28008a8bef34c889998a0b009f0c4/diesel/src/result.rs#L10) so that we can expand this enum in the future, without it being a breaking change?

Reply:
Naming: I went through some pre-PR iteration on this. Calling it SubSecFormat or FractionFormat makes one want to use something like `Omit` for the `Secs` variant and then it doesn't fit together as well.   Its really intended to be one parameter that specifies the seconds format including the `seconds ['.' subseconds]`.  Added more doc to try and clarify this.

If I add `__Nonexhaustive` in this, then I need to add a `panic!()` in `to_rfc3339_opts` to catch the case where it doesn't cover a new variant.   The `std` `ErrorKind` example of this being used is for a case where the enum in question is being returned and consumed by external code.  I don't think there is a valid use case for external code matching on `SecondsFormat` but I added a warning in rustdoc not to do this. I hope that is sufficient.